src/pydvl/utils/parallel/map_reduce.py

```diff
@@ -54,33 +58,42 @@ def get_value(v: Union[ObjectRef, Iterable[ObjectRef], Any], *, timeout: int = 3
 
 
 class MapReduceJob(Generic[T, R]):
-    """Takes an embarrassingly parallel fun and runs it in num_jobs parallel
+    """Takes an embarrassingly parallel fun and runs it in n_jobs parallel
     jobs, splitting the data into the same number of chunks, one for each job.
 
     It repeats the process num_runs times, allocating jobs across runs. E.g.
-    if num_jobs = 90 and num_runs=10, each whole execution of fun uses 9 jobs,
-    with the data split evenly among them. If num_jobs=2 and num_runs=10, two
+    if n_jobs = 90 and num_runs=10, each whole execution of fun uses 9 jobs,
+    with the data split evenly among them. If n_jobs=2 and num_runs=10, two
     cores are used, five times in succession, and each job receives all data.
 
-    Results are aggregated per run using reduce_func(), but not across runs.
+    Results are aggregated per run using `reduce_func`, but **not across runs**.
+    A list of length `n_runs` is always returned.
 
     Typing information for objects of this class requires the type of the inputs
     that are split for `map_func` and the type of its output.
 
     :param map_func: Function that will be applied to the input chunks in each
         job.
     :param reduce_func: Function that will be applied to the results of
-        `map_func` to reduce them.
+        `map_func` to reduce them. This will be done independently for each run,
+        i.e. the reducer need and must not account for data of multiple runs.
     :param map_kwargs: Keyword arguments that will be passed to `map_func` in
-        each job.
+        each job. Alternatively, one can use `itertools.partial`.
     :param reduce_kwargs: Keyword arguments that will be passed to `reduce_func`
-        in each job.
+        in each job. Alternatively, one can use `itertools.partial`.
     :param config: Instance of :class:`~pydvl.utils.config.ParallelConfig`
         with cluster address, number of cpus, etc.
     :param n_jobs: Number of parallel jobs to run. Does not accept 0
-    :param n_runs: Number of times to run the functions on the whole data.
-    :param timeout: Amount of time in seconds to wait for remote results.
-    :param max_parallel_tasks: Maximum number of jobs to start in parallel.
+    :param n_runs: Number of times to run `map_func` and `reduce_func` on the
+        whole data.
+    :param timeout: Amount of time in seconds to wait for remote results before
+        ... TODO
```

Comment:
If I'm not mistaken, a TimeoutError would be raised

Reply:
docs tell me that would be a `ray.exceptions.GetTimeoutError` but also a RayTaskError could be raised here or elsewhere if something fails. So there are many more places to warn about exceptions. Maybe we can delay documenting all exceptions raised to some other time? Also, we would have to abstract the backend somehow.